src/main/java/com/fewlaps/quitnowcache/QNCache.java

```diff
@@ -80,6 +83,33 @@ public void set(String key, T value, long keepAliveInMillis) {
         }
     }
 
+    public List<String> listCachedKeysStartingWith(String keyStartingWith) {
+       	List<String> keys = new ArrayList<String>();
+    	keyStartingWith = getEffectiveKey(keyStartingWith);    	
+
+    	for (String key : Collections.list(cache.keys())) {	
+			if (key.startsWith(keyStartingWith)) {
+    			keys.add(key);
+			}
+    	}
+    	
+    	return keys;
+    }
+    
+    public List<String> listCachedKeysStartingWithIfAlive(String keyStartingWith) {
```

Comment:
Hihi and now, looking at the implementation of Java's Map, I see that Java call this method `keySet()`. What do you think about calling it `keySetStartingWith`? Some thoughts there...